controller/backing_image_controller.go

```diff
@@ -392,171 +475,32 @@ func (bic *BackingImageController) updateDiskLastReferenceMap(backingImage *long
 		},
-		Spec: corev1.PodSpec{
-			ServiceAccountName: bic.serviceAccount,
-			Tolerations:        util.GetDistinctTolerations(tolerations),
-			PriorityClassName:  priorityClass.Value,
-			Containers: []corev1.Container{
-				{
-					Name:            podName,
-					Image:           defaultEngineImage.Value,
-					ImagePullPolicy: imagePullPolicy,
-					Command:         cmdArgs,
-					SecurityContext: &corev1.SecurityContext{
-						Privileged: &privileged,
-					},
-					VolumeMounts: []corev1.VolumeMount{
-						{
-							MountPath:        types.BackingImageDirectoryInContainer,
-							Name:             "backing-image",
-							MountPropagation: &hostToContainerPropagation,
-						},
-					},
-					StartupProbe: &corev1.Probe{
-						Handler: corev1.Handler{
-							Exec: &corev1.ExecAction{
-								Command: []string{
-									"bash", "-c",
-									"ls " + filePathInContainer + " && ls " + downloadedNotificationFlag,
-								},
-							},
-						},
-						// Wait up to 3 hours for the backing image downloading.
-						FailureThreshold:    2160,
-						PeriodSeconds:       5,
-						InitialDelaySeconds: 5,
-					},
-					// Choose ReadinessProbe rather than LivenessProbe due to this Kubernetes bug:
-					// https://github.com/kubernetes/kubernetes/issues/95140
-					ReadinessProbe: &corev1.Probe{
-						Handler: corev1.Handler{
-							Exec: &corev1.ExecAction{
-								Command: []string{
-									"bash", "-c",
-									"ls " + filePathInContainer + " && ls " + downloadedNotificationFlag,
-								},
-							},
-						},
-						PeriodSeconds:       5,
-						InitialDelaySeconds: 5,
-					},
-				},
-			},
-			Volumes: []corev1.Volume{
-				{
-					Name: "backing-image",
-					VolumeSource: corev1.VolumeSource{
-						HostPath: &corev1.HostPathVolumeSource{
-							Path: types.GetBackingImageDirectoryOnHost(diskPath, backingImage.Name),
-						},
-					},
-				},
-			},
-			NodeName:      node.Name,
-			RestartPolicy: corev1.RestartPolicyNever,
+		Spec: types.BackingImageManagerSpec{
+			Image:         defaultImage,
+			NodeID:        node.Name,
+			DiskUUID:      node.Status.DiskStatus[diskName].DiskUUID,
+			DiskPath:      node.Spec.Disks[diskName].Path,
+			BackingImages: requiredBackingImages,
 		},
 	}
+}
 
-	if registrySecret != "" {
-		podManifest.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
-			{
-				Name: registrySecret,
-			},
-		}
-	}
-
-	pod, err := bic.ds.CreatePod(podManifest)
-	if err != nil {
-		if apierrors.IsAlreadyExists(err) {
-			return bic.ds.GetPod(podName)
-		}
-		return nil, err
-	}
-	log := getLoggerForBackingImage(bic.logger, backingImage)
-	log.WithFields(logrus.Fields{"node": node.Name, "diskUUID": diskUUID, "pod": podName}).Infof("Creating download pod for backing image")
-
-	return pod, nil
+func generateBackingImageManagerName(image, diskUUID string) string {
+	return fmt.Sprintf("backing-image-manager-%s-%s", util.GetStringChecksum(image)[:4], diskUUID[:4])
```

Comment:
Can we move this function into types/types.go as a central place for naming stuff?

Reply:
Sure